### PR TITLE
Bug 1851013: Disable creation of External Mode Cluster in cloud environments

### PR DESCRIPTION
**Pull request by bipuladh**
Todo: 
- [ ] Need to verify type for GCP and Azure
Screenshot: 
Screenshot from 2020-09-03 16-03-42

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx [hunk 1]
```diff
@@ -13,14 +14,16 @@ import { OCS_SUPPORT_ANNOTATION, MODES } from '../../constants';
 import { CreateAttachedDevicesCluster } from './attached-devices/install';
 import './install-page.scss';
 
+const CLOUD_PLATFORMS = ['AWS', 'GCP', 'AZURE'];
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx:17]: I will rather publish a whitelist of what's supported.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx [hunk 2]
```diff
@@ -13,14 +14,17 @@ import { OCS_SUPPORT_ANNOTATION, MODES } from '../../constants';
 import { CreateAttachedDevicesCluster } from './attached-devices/install';
 import './install-page.scss';
 
+const EXT_MODE_SUPPORTED_PLATFORMS = ['None', 'VSphere'];
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx:17]:
```suggestion
const EXT_MODE_SUPPORTED_PLATFORMS = ['None', 'VSphere', 'BareMetal'];
```

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx [hunk 3]
```diff
@@ -13,14 +14,17 @@ import { OCS_SUPPORT_ANNOTATION, MODES } from '../../constants';
...
 // eslint-disable-next-line no-shadow
 const InstallCluster: React.FC<InstallClusterProps> = ({ match }) => {
   const {
     params: { ns, appName },
     url,
   } = match;
 
-  const [isIndependent, setIsIndependent] = React.useState(false);
+  const [isIndependent, setIndependent] = React.useState(false);
+  const [isIndepModeSupportedPlatform, setIndepModeSupportedPlatform] = React.useState(false);
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx:27]: Let's keep it as Independent and Converged in the code. Too many places to change

[the remark this answers is not in the document]
Comment (cloudbehl) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx:27]: https://github.com/openshift/console/pull/6507/files#diff-bacf5c497eae04996bf94eab3eab632aR17

then use independent mode here also.

**Pull request merged**